src/commands/ui.rs

```diff
@@ -10,37 +10,66 @@ use ring::{aead, agreement, digest, rand, signature};
 use crate::commands::ExitCode;
 use crate::options::{Options, UI};
 use crate::platform::data_dir;
-use crate::portable::local::{instance_data_dir, InstanceInfo};
+use crate::portable::local::{instance_data_dir, NonLocalInstance};
 use crate::print;
 
 pub fn show_ui(options: &Options, args: &UI) -> anyhow::Result<()> {
     let connector = options.create_connector()?;
     let builder = connector.get()?;
     let mut url = format!("http://{}:{}/ui", builder.get_host(), builder.get_port());
```

Comment:
Only checking `http` looks wrong.

I think we can try `https` first (also we're supposed to have apropriate certificate for that, right?). And only then try http. No? Still not sure whether to print `http` after that (i.e. browser will probably don't have a certificate, but someone may use let's encrypt or corporate certificate that is installed in the browser).

This matters for remote instances. For localhost, any approach is fine (and then always print `http://`)

Reply:
For secure endpoints there will be an http -> https redirect.

Reply:
1. If there is a redirect and reqwest follows redirect it should have a certificate, right? Or will we use redirect only when using globally recognized certificates?
2. Accessing HTTP then HTTPS is suboptimal as it already exposes some info in HTTP mode. So it would be great if we can figure out workflow that tries HTTPS first and only falls back to HTTP when TLS doesn't work for some reason.

Reply:
I'm doing a HTTPS check on remote instances anyways, so that we can get this merged. Please feel free to continue with the discussion.

> Accessing HTTP then HTTPS is suboptimal as it already exposes some info in HTTP mode.
We don't feed auth token to remote instances. Anything else I'm missing here?

Reply:
Yeah, this check is solely to see if the URL in question is accessible, we aren't fetching or sending any data, so I think the HTTP/HTTPS dilemma here is moot.  For maximum compatibility the HTTP endpoints will always support connecting without TLS and will redirect to HTTPS.  Checking for cert validity here also doesn't really achieve much aside from adding friction, because you can "trust" a cert in your browser and it's much harder to do system-wide.